test/remote/totp_tests.js

```diff
@@ -206,6 +208,21 @@ describe('remote totp', function () {
           assert.equal(emailData.headers['x-template-name'], 'newDeviceLoginEmail', 'correct template sent')
         })
     })
+
+    it('should verify totp one outside one code window', () => {
+      const code = otplib.authenticator.generate()
+      return P.delay(4000)
```

Comment:
Looking through previous remote tests, each consistently finishing in under 2 seconds, so having a 4 second delay here seemed like a safe choice for a code window.

Reply:
we cannot wait for a test to be done in 30 secs, even 4 secs is very long time for this test. Can we avoid the 4 secs and test some other way?

Reply:
Generate TOTPs that are 20s ahead and 20s behind window (to check the window is ~90s) and/or 31s ahead/31s behind (to check it fails if outside 90s window), then verify them - should be faster than the 4s delay 

At least, that's what I would do - I don't know if it makes sense for you or not

Reply:
Updated this to remove the delay and check against future and previous code windows.